### DISCUSSION TITLE
Patching packages

### DISCUSSION THREAD

**Discussion by LeCyberDucky** [tectonic-typesetting/tectonic#914] [General]
Hey there,

I sometimes see mentions of patching packages, e.g., when updating to a new bundle version. I was wondering what the scope of this is? 
As an example, I just ran into a very... uh... _interesting_ bug with the `soul` package that will break page numbering in specific scenarios. Miraculously, I managed to come across this fix for the problem, where a simple patch to the package is described: https://tex.stackexchange.com/a/545515/273080

Is applying such fixes within the scope of Tectonic? I spent quite a bit of time figuring this out, so, if possible, it would be nice to save other people from this hassle. But taking care of countless abandoned packages seems like too much to ask from this project.

**Comment by pkgw** [comment]
Thanks for asking! It's a good question. I think that your intuition is correct that it seems like if we go down this road, it's opening up a very big can of worms in terms of maintenance overhead. And there's also the issue of trying to maintain compatibility/exchangeability with TeXLive, which we can view as an upstream for this kind of thing.

I think that the standard that I want to encourage is that we should only patch things when needed to (1) deliver Tectonic-specific feature goals or (2) compensate for something unusual about Tectonic that causes different behavior than standard TeXLive/XeTeX. In other words, we want to produce outputs and behaviors that are identical with TeXLive/XeTeX, *except* in those cases where Tectonic is explicitly trying to do something different.

In your case, for the time being I guess I'd recommend trying to track down the package maintainer and making sure that they're aware of the problem and solution. I have no idea how easy/hard CTAN makes this but get the impression that `soul` is at least actively maintained. For unmaintained/abandoned packages ... well, I think that at some point the Tectonic project might be as much about comprehensive, organized, modern *documentation* for TeX as it is a TeX engine, and in that world I guess the action would be to make sure that the issue with the package is documented. But in that world, maybe we'll have a different relationship with CTAN/TeXLive anyway. It is also interesting to consider a future where packages are pulled from something like a modern crates.io, where package updates can be delivered much more regularly than they are in named TeXLive releases.
